src/pages/Governance/Proposal/card/ParticipationBar.tsx

```diff
@@ -0,0 +1,160 @@
+import React from "react";
+import {Stack, Box, Typography, Tooltip, Theme} from "@mui/material";
+import {secondaryColor} from "../../constants";
+import {useTheme} from "@mui/material/styles";
+import {grey} from "../../../../themes/colors/aptosColorPalette";
+import {useGetCoinSupplyLimit} from "../../hooks/useGetCoinSupplyLimit";
+import {Proposal} from "../../Types";
+
+const RADIUS = "0.7em";
+const BAR_COLOR = secondaryColor;
+const ANCHOR_WIDTH_PERCENTAGE = 1;
+
+function getTotalVotes(proposal: Proposal): number {
+  const yesVotes: number = parseInt(proposal.yes_votes);
+  const noVotes: number = parseInt(proposal.no_votes);
+  return yesVotes + noVotes;
+}
+
+function getFormattedVotesStr(votes: number): string {
+  return votes.toLocaleString("en-US");
+}
+
+function TooltipContent(
+  currentVotes: number,
+  minVotes: number,
+  totalSupply: number,
+): JSX.Element {
+  const currentPercentage = ((currentVotes / totalSupply) * 100).toFixed(0);
+  const minPercentage = ((minVotes / totalSupply) * 100).toFixed(0);
+  const enoughParticipation = currentVotes > minVotes;
+
+  return (
+    <Stack direction="column">
+      <Box>{`- At least ${minPercentage}% participation is needed to proceed with this proposal. `}</Box>
+      <Box>{`- Right now, ${currentPercentage}% of the voting power has participated. `}</Box>
+      <Box>
+        {!enoughParticipation &&
+          `- This proposal will fail if there are not enough votes by the end of the voting period. ${getFormattedVotesStr(
+            minVotes - currentVotes,
+          )} more votes are needed.`}
+      </Box>
+    </Stack>
+  );
+}
+
+function MinVoteAnchor({theme}: {theme: Theme}): JSX.Element {
+  return (
+    <Box
+      component="div"
+      sx={{
+        backgroundColor: theme.palette.mode === "dark" ? grey[50] : grey[800],
+      }}
+      width={`${ANCHOR_WIDTH_PERCENTAGE}%`}
+    />
+  );
+}
+
+type ParticipationBarProps = {
+  proposal: Proposal;
+};
+
+export default function ParticipationBar({proposal}: ParticipationBarProps) {
+  const theme = useTheme();
+
+  const supplyLimit = useGetCoinSupplyLimit();
+  if (!supplyLimit) {
+    return null;
+  }
+
+  const totalSupply = parseInt(supplyLimit);
+  const currentVotes = getTotalVotes(proposal);
+  const minVotes = proposal.min_vote_threshold;
+
+  const currentPercentage = (currentVotes / totalSupply) * 100;
+  const enoughParticipation = currentVotes > minVotes;
+  const part1Percentage = enoughParticipation
```

Comment:
What are all these part1/part2/part3 percentages?

Reply:
I divided the bar in 3 parts, left, middle and right. The left is `Min(currentVotes, minVotes)`; the middle is either `|minVotes-currentVotes|`; and the right is `(totalSupply - Max(currentVotes, minVotes))`.